src/rustdoc-internals/rustdoc-gui-test-suite.md

```diff
@@ -7,6 +7,35 @@ For other rustdoc-specific test suites, see [Rustdoc test suites].
 
 These use a NodeJS-based tool called [`browser-UI-test`] that uses [puppeteer] to run tests in a headless browser and check rendering and interactivity. For information on how to write this form of test, see [`tests/rustdoc-gui/README.md`][rustdoc-gui-readme] as well as [the description of the `.goml` format][goml-script]
 
+### Rustdoc frontend checks in CI (ESLint & TypeScript)
+
+Rustdoc's JS/TS **linting and type-checking** are **not** performed by `compiletest`. In CI they run under the tidy "extra checks" step, which invokes ESLint and `tsc` on rustdoc's frontend sources.
+
+To reproduce locally (mirrors CI behavior for rustdoc's frontend suites):
+
+```bash
+./x.py test --stage 2 tests/rustdoc-js
+./x.py test --stage 2 tests/rustdoc-js-std
+./x.py test --stage 2 tests/rustdoc-gui
```

Comment:
They are not run in any of these cases as far as I know. The only way I know about is:

```suggestion
./x.py test tidy --extra-checks=[js]
```